frontend/src/scene/balance/BalanceView.js

```diff
@@ -100,56 +111,72 @@ class BalanceView extends Component {
+                                        >
+                                            <label htmlFor="amount">
+                                                {i18nContent.amount}
+                                            </label>
+                                            <Input
+                                                name="amount"
+                                                onFocus={
+                                                    this.state.startValidate
+                                                }
+                                                type="number"
+                                                value={this.state.amount}
+                                                onChange={this.setAmount}
+                                            />
+                                        </InputField>
+                                        <ErrorMessage show={paymentFailed}>
+                                            {i18nContent.failureText}
+                                        </ErrorMessage>
+                                    </Form>
+                                    <SubmitButton
+                                        bold
+                                        onClick={this.onConfirm}
+                                        disabled={this.state.formIncorrect}
                                     >
-                                        <label htmlFor="amount">
-                                            {i18nContent.amount}
-                                        </label>
-                                        <Input
-                                            name="amount"
-                                            onFocus={this.state.startValidate}
-                                            type="number"
-                                            value={this.state.amount}
-                                            onChange={this.setAmount}
-                                        />
-                                    </InputField>
-                                </Form>
-                                <SubmitButton
-                                    bold
-                                    onClick={this.onConfirm}
-                                    disabled={this.state.formIncorrect}
-                                >
-                                    {i18nContent.confirm}!
-                                </SubmitButton>
-                            </FormWarpper>
-                        )}
-                    </PoseGroup>
-                </Content>
-            </Modal>
+                                        {i18nContent.confirm}!
+                                    </SubmitButton>
+                                </FormWarpper>
+                            )}
+                        </PoseGroup>
+                    </Content>
+                </Modal>
+
+                <LoadingSpinner
```

Comment:
Loading Spinner could be its own thing. :D

Reply:
changed to a component.